ixwebsocket/IXDNSLookup.cpp

```diff
@@ -27,7 +27,7 @@ namespace ix
         _done(false),
         _id(_nextId++)
     {
-
```

Comment:
I don't know what is the best place to put that init. I'd rather not do it here, but instead at the top of the main.
We would have to call it in the Socket() constructor too, for the people who just want to use a WebSocketServer that doesn't need to do an DNS.

Reply:
Just found this on the internet:

```
Yes, WSAStartup() can be called multiple times in a single process, as long as the requested version is supported by the WinSock DLL. Calls to WSAStartup() and WSACleanup() must be balanced.
```

If we need to balance the number of calls to WSACleanup, then we need a static variable 'initialized' or equivalent, and make sure we only ever call that function once. If we are doing this, it would be good to call call_once().

But this makes me think that it would be best to just *require* that this function is only ever called once at init time, as opposed to have us lazily call it.